apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/PartitionOptions.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using System.Threading;
+
+namespace Google.Cloud.Spanner.Data;
+
+/// <summary>
+/// An immutable class representing partition options.
+/// The specified values of `PartitionSizeBytes`, `MaxPartitions` and `CancellationToken`
+/// are utilized in the RPC call while generating the partition tokens.
+/// The specified flag for `DataBoostEnabled` will be stored in the request property
+/// of the <see cref="CommandPartition"/> object and is used when executing the
+/// partition object. If `DataBoostEnabled` is set to true, the request will be
+/// executed using Spanner independent compute resources.
```

Comment:
Nit: Spanner-independent perhaps? (To be honest, it's not entirely clear to me what that means, but I acknowledge it's probably product wording.)

Reply:
Maybe a link to product documentation here, instead of attempting to explain this?